### PR TITLE
Report errors from curl_easy functions

### PR DESCRIPTION
**Pull request by stweil**
Signed-off-by: Stefan Weil <sw@weilnetz.de>

### CODE REVIEW

**Comment by egorpugin** [comment]
Wow, this definitely must be a lambda.

**Comment by stweil** [comment]
> Wow, this definitely must be a lambda.

I tried that now, but I'm not sure whether it can be simplified further. Suggestions are welcome.

File: src/api/baseapi.cpp [hunk 1]
```diff
@@ -1148,14 +1148,28 @@ bool TessBaseAPI::ProcessPagesInternal(const char* filename,
       return false;
     } else {
       CURLcode curlcode;
+      auto error = [curl, curlcode](const char* function) {
```
Comment (egorpugin) [inline src/api/baseapi.cpp]: Capture `curlcode` by ref? Otherwise it won't be changed.

**Comment by egorpugin** [comment]
Looks better now.

File: src/api/baseapi.cpp (hunk 1 above)
Comment (stweil) [reply src/api/baseapi.cpp]: It isn't changed, so I see no reason to capture it by reference.

Comment (egorpugin) [reply src/api/baseapi.cpp]: I mean it is used in lambda. And you capture some trash stack value.
`curl_easy_strerror(curlcode));`

**Comment by zdenop** [comment]
thanks

**Comment by egorpugin** [comment]
Why? Issue was not fixed.
```CURLcode curlcode; -> you create integer on stack with track in it
auto error = [curl, curlcode](const char* function) -> now you copy that trash into function
```
and what you expect `curl_easy_strerror(curlcode aka TRASH)` will return?

**Comment by zdenop** [comment]
ups. I am sorry - misunderstood your comments...

**Comment by stweil** [comment]
> I mean it is used in lambda. And you capture some trash stack value.

I don't think there is a trash stack value, and the code works as expected.

**Comment by egorpugin** [comment]
Ok, why?

**Comment by stweil** [comment]
The lambda function uses `curl` and `curlcode` from the surrounding function.

**Comment by egorpugin** [comment]
curlcode in the line before lambda is used. Do you agree?

**Comment by stweil** [comment]
That's right, but the lambda function is only used after both local variables have valid values.

**Comment by egorpugin** [comment]
But lambda captures values on its creation right after variable is declared.

**Comment by stweil** [comment]
The compiler agrees:

    baseapi.cpp:1151:27: warning: variable 'curlcode' is uninitialized when used here [-Wuninitialized]

I fixed that now, see commit 9f5a3f6ac790212a124fcbf94413b98a2418a6fd. Thanks for correcting me.

**Comment by egorpugin** [comment]
?????????
It is still wrong, just add & in front of curlcode capture.

Now you will capture only first result and others won't update it again.

**Comment by stweil** [comment]
Thanks again. See commit 4a00b68c6363cb2126be360ff97c1410634108b6.

**Pull request merged**